drivers/drivers.go

```diff
@@ -46,14 +92,86 @@ func NewSession(info *net.OSInfo) OSSession {
 	return nil
 }
 
-func IsOwnExternal(uri string) bool {
-	return IsOwnStorageS3(uri) || IsOwnStorageGS(uri)
-}
-
 func GetSegmentData(uri string) ([]byte, error) {
 	return getSegmentDataHTTP(uri)
 }
 
+// PrepareOSURL used for resolving files when necessary and turning into a URL. Don't use
+// this when the URL comes from untrusted sources e.g. AuthWebhookUrl.
+func PrepareOSURL(input string) (string, error) {
+	u, err := url.Parse(input)
+	if err != nil {
+		return "", err
+	}
+	if u.Scheme == "gs" {
+		m, _ := url.ParseQuery(u.RawQuery)
+		keyfiles, ok := m["keyfile"]
+		if !ok {
+			return u.String(), nil
+		}
+
+		keyfile := keyfiles[0]
+		content, err := ioutil.ReadFile(keyfile)
+		if err != nil {
+			return "", err
+		}
+		u.User = url.User(string(content))
+	}
+	return u.String(), nil
+}
+
+// ParseOSURL returns the correct OS for a given OS url
+func ParseOSURL(input string, useFullAPI bool) (OSDriver, error) {
```

Comment:
false for -objectStore - OS used in transcoding pipeline, true for -recordStore.
This is because of objectStore we need to be able to pass OS credentials between broadcaster and orchestrator and we only need to be able to write segment and to read it back by name.
For recordStore we don't need to pass credentials, but we need to be able list files/dirs, write/read metadata etc.